filestore/handlers.py

```diff
@@ -1,16 +1,62 @@
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
-from .retrieve import HandlerBase
-
 import six
+
 import logging
 import h5py
 import numpy as np
 import os.path
 
+from .retrieve import HandlerBase
+from .readers.spe import PrincetonSPEFile
+
 logger = logging.getLogger(__name__)
 
 
+class IntegrityError(Exception):
+    pass
+
+
+class AreaDetectorSPEHandler(HandlerBase):
+    specs = {'AD_SPE'} | HandlerBase.specs
+
+    def __init__(self, fpath, template, filename,
+                 frame_per_point=1):
+        self._path = fpath
+        self._fpp = frame_per_point
+        self._template = template
+        self._filename = filename
+        self._f_cache = dict()
+
+    def __call__(self, point_number):
+        if point_number not in self._f_cache:
+            fname = self._template % (self._path,
+                                      self._filename,
+                                      point_number)
+            spe_obj = PrincetonSPEFile(fname)
+            self._f_cache[point_number] = spe_obj
+
+        spe = self._f_cache[point_number]
+        data = spe.getData()
+        # if we fix up old entries in CSX FS db we can do va
+        # if data.shape[0] != self._fpp:
+        #     raise IntegrityError(
+        #               "expected {} frames, found {} frames".format(
+        #                      self._fpp, data.shape[0]))
```

Comment:
We can make this change. There is no SPE files which matter in the database at this point